observability/logging/logging-6.0/log6x-clf.adoc

```diff
@@ -5,3 +5,185 @@ include::_attributes/common-attributes.adoc[]
 :context: logging-6x
 
 toc::[]
+
+The `ClusterLogForwarder` CR serves as a single point of configuration for log forwarding, making it easier to manage and maintain log collection and forwarding rules.
+
+* Defines inputs (sources) for log collection
+* Specifies outputs (destinations) for log forwarding
+* Configures filters for log processing
+* Defines pipelines to route logs from inputs to outputs
+* Indicates management state (managed or unmanaged)
+
+// Needs engineering eval re applicability to 6.0
+include::modules/log6x-create-log-forwarder.adoc[leveloffset=+1]
+
+// This needs to be called out extensively.
+.Example ClusterLogForwarder CR
```

Comment:
@jcantrill  @anpingli @r2d2rnd Could you please confirm if we can use the following CLF example in our documentation? If not , could you please provide the example that we can use in our production environment/public docs as is (considering the time frame remaining for Logging 6.0 docs release). That would be great help.

I will add callouts once we have consensus on this example.

I referred [this](https://gitlab.cee.redhat.com/aosqe/aosqe-tools/-/blob/master/logging/log_doc/clf-6.0-examples/clf-multi-forward.yaml?ref_type=heads) link to add the following example.

Reply:
> [This](https://github.com/openshift/cluster-logging-operator/blob/master/docs/reference/log6x-clf-cr-samples.adoc) is an example provided by @libander but it would be worth engineering confirming the server accepts it.

@anpingli Could you please help with your input? Thanks!

Reply:
Log6x-references.adoc had been provided. That is Ok to this CR with a real example.

Reply:
```
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:
  name: logging
  namespace: openshift-logging
spec:
  collector:
    nodeSelector:
      kubernetes.io/os: linux
  filters:
  - name: add-label
    openShiftLabels:
      custom-labels-added: "true"
    type: openShiftLabels
  inputs:
  - application:
      includes:
      - namespace: project-qa-1
      - namespace: project-qa-2
      selector:
        matchLabels:
          test: centos-logtest
    name: selected-app-logs
    type: application
  managementState: Managed
  outputs:
  - lokiStack:
      authentication:
        token:
          from: serviceAccount
      target:
        name: lokistack-sample
        namespace: openshift-logging
    name: default
    tls:
      ca:
        configMapName: openshift-service-ca.crt
        key: service-ca.crt
    type: lokiStack
  pipelines:
  - filterRefs:
    - add-label
    inputRefs:
    - selected-app-logs
    name: pipe1
    outputRefs:
    - default
  serviceAccount:
    name: logcollector
```

Reply:
@libander @smunje1  
The [example ](https://github.com/openshift/cluster-logging-operator/blob/master/docs/reference/log6x-clf-cr-samples.adoc) need to be updated.  

```
  apiVersion: observability.openshift.io/v1
  kind: ClusterLogForwarder
  metadata:
    name: collector # (1)
    namespace: openshift-logging # (2)
  spec:
    inputs:
    - name: mylogqa1
      type: application
      application:
        selector:
          matchLabels:
            test: centos-logtest
        includes:
        - namespace: project-qa-1
        - namespace: project-qa-2
    filters:
    - name: drop-logs
      type: drop
      drop:
      - test:
        - field: .kubernetes.namespace_name
          matches: "openshift*"
    outputs:
    - name: aws-cloudwatch
      type: cloudwatch
      cloudwatch:
        authentication:
          type: accessKey
          awsAccessKey:
            keyID:
              key: aws_access_key_id
              secretName: cloudwatch-credentials
            keySecret:
              key: aws_secret_access_key
              secretName: cloudwatch-credentials
        groupName: <default-value>
        region: us-east-2
    - name: lokistack
      type: lokiStack
      lokiStack:
        authentication:
          token:
            from: serviceAccount
        target:
          name: lokistack-sample
          namespace: openshift-logging
      tls:
        ca:
          key: service-ca.crt
          configMapName: openshift-service-ca.crt
    pipelines:
    - name: all-filter-to-default
      inputRefs:
      - infrastructure
      - application
      - audit
      filterRefs:
      - drop-logs
      outputRefs:
      - lokistack
    - name: selected-app-to-cw
      inputRefs:
      - mylogqa1
      outputRefs:
      - aws-cloudwatch
    serviceAccount:
      name: logcollector
```

Reply:
@anpingli I will add the example you suggested [here](https://github.com/openshift/openshift-docs/pull/80384#discussion_r1721878237) in our documentation. 

```
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:
  name: logging
  namespace: openshift-logging
spec:
  collector:
    nodeSelector:
      kubernetes.io/os: linux
  filters:
  - name: add-label
    openShiftLabels:
      custom-labels-added: "true"
    type: openShiftLabels
  inputs:
  - application:
      includes:
      - namespace: project-qa-1
      - namespace: project-qa-2
      selector:
        matchLabels:
          test: centos-logtest
    name: selected-app-logs
    type: application
  managementState: Managed
  outputs:
  - lokiStack:
      authentication:
        token:
          from: serviceAccount
      target:
        name: lokistack-sample
        namespace: openshift-logging
    name: default
    tls:
      ca:
        configMapName: openshift-service-ca.crt
        key: service-ca.crt
    type: lokiStack
  pipelines:
  - filterRefs:
    - add-label
    inputRefs:
    - selected-app-logs
    name: pipe1
    outputRefs:
    - default
  serviceAccount:
    name: logcollector
```

Cc @libander

Reply:
I have added the above example as part of this PR. 

@libander will cover the callouts for this example as part of the larger PR (The larger PR, which will be created before Logging 6.0 release)